pkgs/applications/networking/cluster/helm/default.nix

```diff
@@ -1,39 +1,39 @@
-{ stdenv, fetchurl, kubectl }:
-let
-  isLinux = stdenv.isLinux;
-  arch = if isLinux
-         then "linux-amd64"
-         else "darwin-amd64";
-  checksum = if isLinux
-             then "18bk4zqdxdrdcl34qay5mpzzywy9srmpz3mm91l0za6nhqapb902"
-             else "03xb73769awc6dpvz86nqm9fbgp3yrw30kf5lphf76klk2ii66sm";
-  pname = "helm";
+{ stdenv, buildGoPackage, fetchgit }:
+
+buildGoPackage rec {
   version = "2.11.0";
-in
-stdenv.mkDerivation {
-  name = "${pname}-${version}";
+  name = "helm-${version}";
 
-  src = fetchurl {
-    url = "https://kubernetes-helm.storage.googleapis.com/helm-v${version}-${arch}.tar.gz";
-    sha256 = checksum;
+  src = fetchgit {
+    rev = "v${version}";
+    url = "https://github.com/helm/helm.git";
+    sha256 = "1z810a6mxyrrw4i908dip8aqsj95c0kmv6xpb1wwhskg1zmf85wk";
   };
 
-  preferLocalBuild = true;
+  goPackagePath = "k8s.io/helm";
+  subPackages = [ "cmd/helm" "cmd/tiller" "cmd/rudder" ];
+
+  goDeps = ./deps.nix;
 
-  buildInputs = [ ];
+  # this is to get buildGoPackage to output to the standard $out folder
+  outputs = [ "out" ];
 
-  propagatedBuildInputs = [ kubectl ];
+  preBuild = ''
```

Comment:
It would better to use the `buildFlagsArray` attribute such as in https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/skopeo/default.nix#L33 for instance.

Reply:
I agree - though I couldn't get that to work before. I copied this from another chart and it worked.